tests/unit/cpp/test_inet.cpp

```diff
@@ -318,273 +318,314 @@ TEST(etcpal_cpp_inet, ip_from_string_works)
-    TEST_ASSERT_TRUE(sockaddr4.ip().IsValid());
-    TEST_ASSERT_TRUE(sockaddr4.ip().IsV4());
-    TEST_ASSERT_EQUAL_UINT32(sockaddr4.ip().v4_data(), 0xdeadbeef);
-    TEST_ASSERT_EQUAL_UINT16(sockaddr4.port(), 0xfeed);
+  // Construct from raw IPv4 data and port number
+  const etcpal::SockAddr sockaddr1(0xdeadbeef, 0xfeed);
+  TEST_ASSERT_TRUE(sockaddr1.IsValid());
+  TEST_ASSERT_TRUE(sockaddr1.IsV4());
+  TEST_ASSERT_EQUAL_UINT32(sockaddr1.v4_data(), 0xdeadbeef);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr1.port(), 0xfeed);
+
+  // Construct from raw IPv6 data and port number
+  const etcpal::SockAddr sockaddr2(s_v6_data.data(), 0xbeef);
+  TEST_ASSERT_TRUE(sockaddr2.IsValid());
+  TEST_ASSERT_TRUE(sockaddr2.IsV6());
+  TEST_ASSERT_EQUAL_UINT8_ARRAY(sockaddr2.v6_data(), s_v6_data.data(), ETCPAL_IPV6_BYTES);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr2.port(), 0xbeef);
+
+  // Construct from raw IPv6 data, a scope ID and port number.
+  const etcpal::SockAddr sockaddr3(s_v6_data.data(), 12, 0xdead);
+
+  TEST_ASSERT_TRUE(sockaddr3.IsValid());
+  TEST_ASSERT_TRUE(sockaddr3.IsV6());
+  TEST_ASSERT_EQUAL_UINT8_ARRAY(sockaddr3.v6_data(), s_v6_data.data(), ETCPAL_IPV6_BYTES);
+  TEST_ASSERT_EQUAL(sockaddr3.scope_id(), 12);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr3.port(), 0xdead);
+
+  // Construct from an IpAddr and port number
+  const etcpal::SockAddr sockaddr4(etcpal::IpAddr(0xdeadbeef), 0xfeed);
+  TEST_ASSERT_TRUE(sockaddr4.IsValid());
+  TEST_ASSERT_TRUE(sockaddr4.IsV4());
+  TEST_ASSERT_EQUAL_UINT32(sockaddr4.v4_data(), 0xdeadbeef);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr4.port(), 0xfeed);
 }
 
 TEST(etcpal_cpp_inet, sockaddr_to_string_works)
 {
-    const etcpal::SockAddr sockaddr_v4(etcpal::IpAddr::FromString("10.101.2.3"), 5555);
-    TEST_ASSERT_EQUAL_STRING(sockaddr_v4.ToString().c_str(), "10.101.2.3:5555");
+  const etcpal::SockAddr sockaddr_v4(etcpal::IpAddr::FromString("10.101.2.3"), 5555);
+  TEST_ASSERT_EQUAL_STRING(sockaddr_v4.ToString().c_str(), "10.101.2.3:5555");
 
-    const etcpal::SockAddr sockaddr_v6(etcpal::IpAddr::FromString("2001:db8::2222:3333"), 6666);
-    auto sockaddr_str = sockaddr_v6.ToString();
-    ConvertStringToLowercase(sockaddr_str);
-    TEST_ASSERT_EQUAL_STRING(sockaddr_str.c_str(), "[2001:db8::2222:3333]:6666");
+  const etcpal::SockAddr sockaddr_v6(etcpal::IpAddr::FromString("2001:db8::2222:3333"), 6666);
+  auto sockaddr_str = sockaddr_v6.ToString();
+  ConvertStringToLowercase(sockaddr_str);
+  TEST_ASSERT_EQUAL_STRING(sockaddr_str.c_str(), "[2001:db8::2222:3333]:6666");
 
-    const etcpal::SockAddr sockaddr_invalid;
-    TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+  const etcpal::SockAddr sockaddr_invalid;
+  TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+}
+
+TEST(etcpal_cpp_inet, sockaddr_is_link_local_works)
+{
+  const auto sockaddr = etcpal::SockAddr(etcpal::IpAddr::FromString("169.254.20.40"), 8888);
+  TEST_ASSERT_TRUE(sockaddr.IsLinkLocal());
+}
```

Comment:
Do we need to TEST_ASSERT_FALSE for some of the other types (e.g. IsMulticast)?

Reply:
I've added your requested tests on the develop branch in 2eb339c.